docs/source/notes/autograd.rst

```diff
@@ -598,3 +643,83 @@ chain rule:
 
         .. math::
             \frac{\partial L}{\partial z^*} = 2 * Re(grad\_out^* * \frac{\partial s}{\partial z^{*}})
+
+.. _saved-tensors-hooks-doc:
+
+Hooks for saved tensors
+-----------------------
+
+You can control :ref:`how saved tensors are packed / unpacked
+<saved-tensors-doc>` by defining a pair of ``pack_hook`` / ``unpack_hook`` hooks.
+The ``pack_hook`` function should take a tensor as its single argument but can
+return any python object (e.g. another tensor, a tuple, or even a filename).
+The ``unpack_hook`` function takes as its single argument the output of
+``pack_hook`` and should return the original tensor to be used in the backward
```

Comment:
"the original tensor" -> we should define that better here.
It can actually be a different Tensor object. It can have a different storage, etc

The requirement is that it should be a Tensor that has the same content.
Any autograd-related metadata should be ignored in particular as they will be overwritten.